### PR TITLE
Added social card page and puppeteer for taking screenshots and storing them

### PR DESCRIPTION
**Pull request by SuragNuthulapaty**
## Overview

Creates the social-card page in order to make the meta images

Adresses #97 

### Demo
Examples of what the social cards look like (note: my screenshots are a bit large, the social cards are the proper size)
Screen Shot 2022-08-03 at 6 01 18 PM
Screen Shot 2022-08-03 at 6 01 45 PM

## Testing Instructions
you can see more states by going to /social-card?state={name of state}

### CODE REVIEW

**Comment by vkoves** [comment]
@SuragNuthulapaty - CI is failing, try running the lint fix in the README like we ran last night. Also just overall amazing work, I'm so impressed that you already got this done!

There's also an issue in the Netlify build with using `document.location` - check state-details, maybe it's just `location` 🤔 

image

**Comment by SuragNuthulapaty** [comment]
@vkoves I've run the lint and it still seems to not want to work. Also, in state-details we already filter the data for the state so it doesn't use the `document.location` so I'm not sure where to go from there

**Comment by vkoves** [comment]
@SuragNuthulapaty - it just just be a build issue then. Search for location again though, we extracted the code from state details since it does already have filtered data but it uses the URL to get the state name

**Comment by SuragNuthulapaty** [comment]
@vkoves state-details doesn't have any use of "URLSearchParams" and its two references to location is from a parameter passed into the function that gives the pathname of the page. We can't use this same approach because the pathname for passing in the location comes to just `/social-card` and doesn't have the `?state=illinois` for example. If there is a way to get the second part of the link, I can't figure it out. 

(Note: just /social-card is from the location parameter, I added the location to denote which console.log I was looking for)
Screen Shot 2022-08-04 at 7 15 38 AM

**Comment by vkoves** [comment]
It looks like with the actual Gatsby prod build the Illinois icon is always used, and the graph doesn't load, check out this [test url](https://deploy-preview-137--decarb-my-state.netlify.app/social-card/?state=oklahoma):

image

**Comment by SuragNuthulapaty** [comment]
Weird, I can't seem to replicate this in the code I have on the branch. Maybe it is a netlify issue?

**Comment by vkoves** [comment]
@SuragNuthulapaty - it'll probably be easier to walk through this on our call on Tuesday, but a local dev build works differently than the actual prod build. I forget how you run a prod build locally, so I think we'll need Derek's help to debug this.

File: src/generate-social-cards.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ * This file lops through every state and takes screenshots of the social card for each state
+ * This script is only meant to be run locally
+ */
+
+import { launch } from "puppeteer"
+import { existsSync, mkdirSync } from "fs"
+
+var waitingForReady = true
+
+const placeNames = [
```
Comment (vkoves) [inline src/generate-social-cards.js:11]: Make sure to split this into a constant. Maybe `src/constants/state-names`?

File: src/generate-social-cards.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  "virginia",
+  "washington",
+  "west_virginia",
+  "wisconsin",
+  "wyoming",
+]
+
+/*
+ * For each state, use puppeteer to capture a screnshot
+ * @param name {strign} the name of the state to have the social-card taken a picture of
```
Comment (vkoves) [inline src/generate-social-cards.js:67]: Typo: `strign` - `string`.

File: src/generate-social-cards.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+//wait for the previous state to finish taking screenshot before making the next one
+setTimeout(async function () {
+  for (const name of placeNames) {
+    while (!waitingForReady) {
+      setTimeout(function () {}, 1000)
+    }
+    await captureScreenshot(name)
+  }
+}, 3000)
```
Comment (vkoves) [inline src/generate-social-cards.js:110]: There's an idea in code about avoiding magic strings and magic numbers, which are numbers in code that are uncelar. It's best to move this to a constant, like `const SocialPageDelayMs`. Similarly for the `1000` you can call that something like `WaitDelayMs`.

File: src/styles/social-card.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+.social-card h2 {
+    background-color: #ff5722;
+    color: white;
+}
+
+.social-card h3 {
+    font-size: 3rem;
+}
```
Comment (vkoves) [inline src/styles/social-card.css:13]: Add a newline to end of files - it's kind of an old thing to work with automated file reading systems, but it's a good practice and you can see GitHub flags it up.

**Review by vkoves — CHANGES_REQUESTED** [review]
Have a few minor starter comments, also noticed the emoji have no spacing with the text:

Screenshot from 2022-08-09 21-26-12

Since space is at a premium, I wonder if we should reduce the font-size of the state icons since states like Connecticut are quite wide, and the visual is more cosmetic than anything.

**Comment by SuragNuthulapaty** [comment]
@vkoves i already reduce the font size, though I can do it a bit more

**Review by vkoves — APPROVED** [review]
Looks great, awesome work! I skimmed all of the images to make sure no text was broken, and also confirmed all the pages still work properly.

**Pull request merged**